### PR TITLE
test: cover simtest state and admin API

### PR DESCRIPTION
**Pull request by Peyton-Spencer**
## Summary
- add deterministic coverage for `FakeState` task, message, IPC, avatar, and reset mutations used by the simtest web harness
- add admin API request tests for task lifecycle, IPC validation, remote peer logs, scenarios, CORS preflight, and invalid JSON handling
- add supporting coverage for `src/ipc.ts` ownership/quarantine behavior and Telegram avatar helper parsing/sanitization
- resolve CI blockers by replacing the Bun-incompatible `TimerHandler` mock type, tightening a peer-log assertion to use a delta, and formatting the changed tests

## Testing
- `bun test src/ipc-watcher.test.ts src/simtest/admin-api.test.ts src/simtest/fake-state.test.ts src/telegram-avatar.test.ts`
- `bunx tsc --noEmit`
- `bunx prettier --check "src/ipc-watcher.test.ts" "src/simtest/admin-api.test.ts" "src/simtest/fake-state.test.ts" "src/telegram-avatar.test.ts"`

## Coverage Notes
- Files covered in this PR: `src/simtest/fake-state.ts`, `src/simtest/admin-api.ts`, `src/ipc.ts`, `src/telegram-avatar.ts`
- Remaining notable gaps: `src/index.ts`, `src/whatsapp-auth.ts`

### CODE REVIEW

File: src/simtest/fake-state.test.ts [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      true,
+    );
+
+    const filtered = state.searchMessages('pr #315', 'sim:general', 1);
+    expect(filtered).toHaveLength(1);
+    expect(filtered[0]?.chat_jid).toBe('sim:general');
+
+    expect(state.searchMessages('   ')).toEqual([]);
+  });
+
```
Comment (Peyton-Spencer) [inline src/simtest/fake-state.test.ts]: The `searchMessages` test exercises whitespace trimming (`"  pr #315  "`), cross-channel search, limit clamping, and empty-query-returns-empty. Thorough coverage of the search boundary conditions.
